### ISSUE TITLE
Feature Request: Disallow the taking of screenshots of lockscreen.

### ISSUE THREAD

**Issue by raccspoons** [ubports/ubuntu-touch#1015]
I am using a OPO (stable, OTA-5) and was also experiencing this issue on a Nexus 5 (stable, OTA-4).

In short, I am generating about 2-15 unintentional screenshots per day. What follows may be an opinion, but having to press two buttons that are next to and are ATTACHED to each other will intuitively result in this problem. SailfishOS uses the same button combo, so maybe there is some annoying trademark or patent issue that I am not thinking about that is the reason this particular decision was made. The number of screenshots are small, so this is nothing but an annoyance.

As no one is complaining about this issue, and I have experienced it frequently on two UT devices, maybe I am just using my phone in a really wacky way. It usually happens as I am fumbling with the phone as I take it out of my pocket, when I pull it out upsidedown or when I am carrying the phone in my hand while trying to do other things. I also have a lot of screenshots of times when I have attempted to change the volume.

I use a two-layer case, that has covered buttons which is likely making the issue worse, but it also happens without a case on both devices. I have also **NEVER** unintentionally taken a screenshot on any smartphone running any other OS with any case.

Short of changing the button combination to ANYTHING else (Power + Vol Up?, Power + Vol Down?, Vol Up + Vol Down + Finger on screen?), the thing that would solve the bulk of this problem would be to disallow the taking of a screenshot while the phone is locked. I for one will never in a million years INTEND to take a screenshot of my lockscreen. There may be reasons to, but I can't think of one

**Comment by NeoTheThird** [comment]
I agree, the average user probably never does it. I have actually taken quite a few lock-screen screenshots for bug reports or promotional material though, so if we do this, i'd definitely want an option to turn screenshots of the lock-screen back on.

**Comment by raccspoons** [comment]
Changing the button combination to Volume Up + Power (or anything else) would be the most parsimonious way to fix all of the problems, but as I'm not a developer, I have no idea how difficult that would be. Re-training all of the users is probably the insurmountable obstacle for such a limited benefit.

The UT Tweak tool or buried deep in the settings seems like the place to put a feature that users do not use, and that is only used by members/developers of this project and/or for filing rare bugs that affect the appearance of the lock screen and that cannot be described with words. But again, maybe users are using this feature all the time? Maybe it just doesn't bother most people? Or maybe the way I use my phone with the case I am using is resulting in this happening way more often for me than normal users? In general, it is probably better if a device doesn't have a hair-trigger to unintentionally take photos of one's activities and store them unencrypted in a way that the user doesn't necessarily know that they are there.

**Comment by dobey** [comment]
I must say, I have never accidentally taken a screenshot with Ubuntu, on my Nexus 5, which itself is a device notorious for faulty power and volume buttons. That is from a span of over 2-2.5 years as my only phone and only phone OS. I've also never heard or seen any similar complaint from anyone else, in the 5+ years that Ubuntu has been available for phones/tablets. So, I am personally quite curious as to how you've actually managed to do this accidentally, and so often.

**Comment by raccspoons** [comment]
Good to know, but we are different people with different lives that use their technology in different ways. It is not the least bit surprising that someone wouldn't have this issue, but what is relevant is that I AM. 

I am a fairly "active" phone user. I listen to a lot of podcasts throughout the day, so I am constantly pulling my phone out of my pocket to pause, unpause, change podcast, etc. I wear a lab coat and pants on the tighter side. The biggest offender is that I use a case with covered, protruding buttons (most of the OPO cases are this style). These are all making the issue worse, but these are all things that an _actual user_ in the world might do. Also, the cases for these devices were not designed for UT devices with this button combination restriction, so a case with cutouts for the buttons or no case at all would probably work better.

When the iPhone 4 came out, it was dropping calls when the user touched the side of the phone, and the official response from Apple was that "No solution is required because those people are just idiots holding the phone incorrectly." followed by a well-earned backlash.

If you can't imagine how someone could accidentally press the Vol. Up and Vol. Down button at the same time, then you are not very imaginative. If I don't happen to use my device exactly like at least one of the developers on a very small team, a feature request is the only way to let them know that people like me exist. Telling me you are not having this issue is not interesting, as obviously, the vast majority of users are not having this problem. As the user-base expands, you are going to add more people like me to it. Apple and Google have solved this years ago for the users like me by choosing a button combination that requires more intentionality to press.

**Comment by dobey** [comment]
There is no need for you to respond in such manner to my simply stating that I have never had this experience, nor have I heard of anyone else having such experience before, and stating that I am curious as to how it actually happens.

What I want to do is to fully understand the problem, so that a proper solution can be made, to make the experience better for everyone, not just you. Condescendingly demanding a specific solution without any understanding of how the problem is occurring for you, but not for everyone else, is inappropriate.

**Comment by raccspoons** [comment]
Yeesh. Maybe read it again?? If you look for the part where I have made anything even resembling a demand, you will not find it. I PERSONALLY do not care if this ever gets implemented, nor do I expect it to be. This problem would be solved for ME by buying a case with cutouts for the buttons. I'm not even bothered enough to do that. My pain-tolerance for buggy software is very high, and this issue is the minorest-of-all-minor-annoyances. However, I have noticed it, which means a new user that may not be as pain-tolerant as I am might also notice. As a non-developer, I feel actual guilt that I cannot contribute in more meaningful ways, so the ways in which I can contribute are financial (I am a Patron) and through issue-reporting. Google and Apple have whole teams doing nothing but thinking about their UI, and this project does not. There are many rough edges I have noticed with the UI, and it is my intention to report ones that have not been reported yet, and to propose possible solutions.

Until you changed the title of this issue, I would have had no idea that you were associated with this project in any way. Your bio does not mention it, nor do you have a Member tag by your name. I don't know ANYTHING about you. So you're just some user that entered this conversation to say you don't have this issue, and you can't understand how anyone could. If you are developer on a fact-finding mission, asking an a QUESTION would be a way to convey this. I pushed back and explained why I was pushing back, but also was not annoyed and didn't have a problem in any way with what you said, nor did I blow up on you like you have done to me.

This is NOT a software bug. The fact that you cannot reproduce it is not interesting or relevant. NOT because the things you say are not interesting, but because this is an issue stemming from the way I have trained myself to hold a smartphone over the past decade. This could be your only phone OS for 2 years. 10 years. 20 years. It doesn't matter, YOU will never have this issue, because YOU don't use the device in the same way. However, I DO. I am attempting to build the case that, even if none of the developers hold their phone in the same way, that my behavior IS within the realm of normal, so IS worth considering. If you actually look, you'll see I am attempting to provide a description of this issue and mountains of context you claim to desire.

I've been using UT as a daily-driver for just over a month, and have been excited about this project in a way I haven't been about one in a long time. I have nothing but gratitude and respect for the contributors for this project. Accusing me of being the loathsome, entitled newbie for making a good-intentioned feature request is pretty demoralizing.

**Comment by Flohack74** [comment]
@raccspoons Please keep your comments short and simple to read. This is a bug tracker, if you want to actually discuss the worthiness of this or that solution please use the forum http://forums.ubports.com . If I have to read a wall of text to come to the point, I am not very motivated to even start thinking how to help you.

**Comment by raccspoons** [comment]
Got it. Sorry for wasting everyone's time.

**Comment by kugiigi** [comment]
I also have this problem with my Nexus 5 and it's really quite often.
But I don't have this problem with my bq E5.
So I think it's really just the Nexus 5 having a bad placement of the volume buttons and the power button. Add to that the my power button becomes harder and harder to press so the tendency is to press the volume buttons instead.

I'm not sure but from what I see in images, OPO has the same button placement.
I think the current screenshot shortcut is fine although changing it to Power + Volume Down/Up will lessen the tendency of accidental trigger. (It'll be a disaster for Nexus 5 owners though 😜 )

**Comment by ruralelectrification** [comment]
I also have this one (OnePlus One), and I think it is pretty weird that this isn't more widespread. I have a chunky case, but it happens without one.

There are two main situations where this happens for me:
1) When I want to turn the volume down QUICKLY because some video started blaring unexpectedly. I end up desperately mashing the volume down button, but hitting both.
2) Pushing the power button requires more force to press than the weight of the phone, so I need to brace the other side. The volume buttons are directly across, so sometimes end up getting pressed at the moment the screen wakes up, taking a screenshot. This situation is impossible in iOS/Android, so I've never had to think about being ultra careful with pressing buttons. I don't really know how to untrain myself, but I suppose I will eventually? 

Disabling screenshot ability on lockscreen would be good, but I also think changing the button combination would be a disaster for current users.

**Comment by ruralelectrification** [comment]
Should this be marked as an opinion?

**Comment by ruedigerkupper** [comment]
I also quite frequently produce unwanted screen shots, on my Xperia X and formerly on my Nexus 5. It is just very easy to hit both buttons at once – or even confuse them with the power button. However, power + vol up is not a good choice for an alternative. At least on the Xperia X, this combination results in hard power off of the device.
